cli/azd/pkg/cache/file_cache.go

```diff
@@ -0,0 +1,114 @@
+package cache
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/osutil"
+)
+
+// CacheResolver is a function that resolves the cache value.
+type CacheResolver[T any] func(ctx context.Context) (*T, error)
+
+// FileCache is a cache that stores the value in a file otherwise resolves it.
+type FileCache[T any] struct {
+	filePath      string
+	resolver      CacheResolver[T]
+	cacheDuration time.Duration
+	value         *T
+}
+
+// NewFileCache creates a new file cache.
+func NewFileCache[T any](cacheFilePath string, cacheDuration time.Duration, resolver CacheResolver[T]) *FileCache[T] {
```

Comment:
How is this FileCache used for the extension commands?   Should it be a different PR?  What's the relation?  I only see it used from tests